src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -413,6 +433,35 @@ begin
   exact pi_ne_zero
 end
 
+@[simp] lemma to_real_pi_div_two : ((π / 2 : ℝ) : angle).to_real = π / 2 :=
+begin
+  rw [to_real_coe_eq_self_iff],
+  exact ⟨(left.neg_neg_iff.2 real.pi_pos).trans (div_pos real.pi_pos two_pos),
+         div_le_self real.pi_pos.le one_le_two⟩
+end
+
+@[simp] lemma to_real_neg_pi_div_two : ((-π / 2 : ℝ) : angle).to_real = -π / 2 :=
+begin
+  rw [to_real_coe_eq_self_iff],
+  refine ⟨_,
+          (div_nonpos_of_nonpos_of_nonneg (left.neg_neg_iff.2 real.pi_pos).le two_pos.le).trans
+             real.pi_pos.le⟩,
+  rw neg_div,
+  exact neg_lt_neg_iff.2 (div_lt_self real.pi_pos one_lt_two)
+end
+
+lemma pi_div_two_ne_zero : ((π / 2 : ℝ) : angle) ≠ 0 :=
```

Comment:
Can you prove something like `half_ne_zero`?

Reply:
The condition for half a real to be nonzero as an angle is that the real isn't an integer multiple of `4 * π`. I don't think proving something isn't an integer multiple of `4 * π` is very convenient (based on experience, when I tried to prove `real.angle.pi_ne_zero` based on not being an integer multiple of `2 * π`, and it was certainly a lot more complicated than the version based on `to_real`).

Reply:
I meant an implication, not an `iff`.

Reply:
I'm still not clear on what you think would be useful here (given these are two-line proofs, so it's hard to be much simpler). And I suspect proving some other concrete angle is nonzero is more likely to arise than proving the result of dividing a real by 2 is a nonzero angle, but it's hard to tell what general result might be useful in future without more use cases.

Reply:
Probably you're right.